ui/packages/catalog-ui-search/src/main/webapp/component/search-interactions/search-interactions.view.js

```diff
@@ -36,29 +33,37 @@ module.exports = Marionette.LayoutView.extend({
         searchType: '.interaction-type',
         resultType: '.interaction-result-type',
         searchAdvanced: '.interaction-type-advanced',
+        customInteractions: '.interaction-custom',
         searchSettings: '.interaction-settings'
     },
     events: {
         'click > .interaction-reset': 'triggerReset',
         'click > .interaction-type-advanced': 'triggerTypeAdvanced'
     },
-    onRender: function(){
+    initialize() {
+        this.listenTo(this.model, 'change:type', this.render);
+    },
+    onRender(){
         this.listenTo(this.model, 'change:type closeDropdown', this.triggerCloseDropdown);
         this.generateSearchFormSelector();
         if(properties.hasExperimentalEnabled()) { 
             this.generateResultFormSelector() 
         }
+        if(this.model.get('type') === 'custom') {
+            this.generateCustomFormSelector();
+            this.customInteractions.$el.toggleClass('is-hidden', false);
```

Comment:
✏️ Typically we avoid toggling classes on elements within the HTML and stick to toggling them on the main root el (`this.$el`).  In this case, you could toggle a class called `is-custom` depending on the model's type.